subprojects/ide/src/main/java/org/gradle/plugins/ide/internal/resolver/IdeDependencySet.java

```diff
@@ -201,21 +206,31 @@ private void resolveAuxiliaryArtifacts(IdeDependencyVisitor visitor) {
 
         private void visitArtifacts(IdeDependencyVisitor visitor) {
             for (ResolvedArtifactResult artifact : resolvedArtifacts.values()) {
-                ComponentIdentifier componentIdentifier = artifact.getId().getComponentIdentifier();
+                ComponentArtifactIdentifier artifactIdentifier = artifact.getId();
+                ComponentIdentifier componentIdentifier = artifactIdentifier.getComponentIdentifier();
                 if (componentIdentifier instanceof ProjectComponentIdentifier) {
                     visitor.visitProjectDependency(artifact);
                 } else if (componentIdentifier instanceof ModuleComponentIdentifier) {
                     Set<ResolvedArtifactResult> sources = auxiliaryArtifacts.get(componentIdentifier, SourcesArtifact.class);
                     sources = sources != null ? sources : Collections.<ResolvedArtifactResult>emptySet();
                     Set<ResolvedArtifactResult> javaDoc = auxiliaryArtifacts.get(componentIdentifier, JavadocArtifact.class);
                     javaDoc = javaDoc != null ? javaDoc : Collections.<ResolvedArtifactResult>emptySet();
-                    visitor.visitModuleDependency(artifact, sources, javaDoc);
+                    visitor.visitModuleDependency(artifact, sources, javaDoc, isTestConfiguration(configurations.get(artifactIdentifier)));
                 } else {
-                    visitor.visitFileDependency(artifact);
+                    visitor.visitFileDependency(artifact, isTestConfiguration(configurations.get(artifactIdentifier)));
                 }
             }
         }
 
+        private boolean isTestConfiguration(Set<Configuration> configurations) {
+            for (Configuration c : configurations) {
+                if (!c.getName().toLowerCase().contains("test")) {
+                    return false;
+                }
+            }
+            return true;
+        }
```

Comment:
It's unfortunate that this is based solely on the name rather than a property of the configuration or that the configuration extends test. My org has hundreds of projects that have an 'acceptance' configuration which extends test. I am one of a few eclipse users and can't rightly change the "standard" name from "acceptance" to "acceptanceTest" to make Gradle play nice with Eclipse. Has this considered that our "acceptance" configuration extended "test" it would configure the projects properly. Or less brittle, if the configurations had a test attribute that could be set, the default "test" configuration included, the configurations could work correctly without the magical name.contains that could cause false positives with some name like 'contestant'.

Reply:
Thanks for the explanation, makes sense to me. Would you be interested providing a pull request for this?

Reply:
I can try. It's rather annoying to fix it every time I refresh a project.

This init.gradle script has kept me sane but doesn't work for some multi-module projects:
```
allprojects {
    if (project.plugins.hasPlugin(EclipsePlugin)) {
        eclipse.synchronizationTasks {
            eclipse.classpath.file.whenMerged {
                entries.findAll { entry ->
                    entry.class.name in [
                        'org.gradle.plugins.ide.eclipse.model.Library',
                        'org.gradle.plugins.ide.eclipse.model.SourceFolder'
                    ]
                }.each { entry ->
                    if (entry.entryAttributes['gradle_used_by_scope'] == 'acceptance') {
                        entry.entryAttributes['test'] = 'true'
                    }
                }
            }
        }
    }
}
```